lib/providers/network_providers/login_provider.dart

```diff
@@ -0,0 +1,29 @@
+import 'package:cobe_hive_mobile_app/data/models/login_response.dart';
+import 'package:cobe_hive_mobile_app/data/repository/login_repository.dart';
+import 'package:cobe_hive_mobile_app/providers/login_screen_providers/email_provider.dart';
+import 'package:cobe_hive_mobile_app/providers/login_screen_providers/password_provider.dart';
+import 'package:cobe_hive_mobile_app/providers/network_providers/login_repository_provider.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+final loginProvider =
+    StateNotifierProvider<LoginNotifier, LoginResponse>((ref) {
+  return LoginNotifier(
+    ref.read(loginRepositoryProvider),
+    ref.watch(emailProvider),
+    ref.watch(passwordProvider),
```

Comment:
You need to be carefull when a statenotifier provider rebuilds on other providers.

It is better to pass a reference to a `Ref` object to `LoginNotifier`and in `login()` you can `read(emailProvider)` and `read(passwordProvider)`